gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/BaseDistributionExtensionTest.groovy

```diff
@@ -46,87 +44,68 @@ class BaseDistributionExtensionTest extends Specification {
         Project child = ProjectBuilder.builder().withName("child").withParent(parent).build()
 
         then:
-        new BaseDistributionExtension(child).serviceName == "child"
+        new BaseDistributionExtension(child, child.getObjects()).serviceName.get() == "child"
     }
 
     def 'serviceGroup uses project group as default'() {
         when:
-        Project project = ProjectBuilder.builder().build()
         project.group = "foo"
 
         then:
-        new BaseDistributionExtension(project).serviceGroup == "foo"
+        new BaseDistributionExtension(project, project.getObjects()).serviceGroup.get() == "foo"
     }
 
     def 'serviceGroup can be overwritten'() {
         when:
-        Project project = ProjectBuilder.builder().build()
         project.group = "foo"
 
         then:
-        def ext = new BaseDistributionExtension(project)
-        ext.serviceGroup("bar")
-        ext.serviceGroup == "bar"
-    }
-
-    def 'productType only accepts valid values'() {
-        when:
-        def ext = new BaseDistributionExtension(null)
-        ext.productType "foobar"
-
-        then:
-        def ex = thrown IllegalArgumentException
-        ex.message == "Invalid product type 'foobar' specified; supported types: [service.v1, daemon.v1, asset.v1, pod.v1]."
-    }
-
-    def "productDependencies from invalid maven coordinate"() {
-        when:
-        def ext = new BaseDistributionExtension(null)
-        ext.productDependency("group:name")
-
-        then:
-        thrown(IllegalArgumentException)
-    }
-
-    def "productDependencies from maven coordinate and no recommended version"() {
-        when:
-        def ext = new BaseDistributionExtension(null)
-        ext.productDependency("group:name:1.2.3")
-
-        then:
-        ext.serviceDependencies.size() == 1
-        ext.serviceDependencies.get(0).productGroup == "group"
-        ext.serviceDependencies.get(0).productName == "name"
-        ext.serviceDependencies.get(0).minimumVersion == "1.2.3"
-        ext.serviceDependencies.get(0).maximumVersion == "1.x.x"
-        ext.serviceDependencies.get(0).recommendedVersion == null
+        def ext = new BaseDistributionExtension(project, project.getObjects())
+        ext.setServiceGroup("bar")
+        ext.serviceGroup.get() == "bar"
     }
 
-    def "productDependencies from maven coordinate with all fields and no recommended version"() {
+    // TODO(forozco): verify that enums work correctly
+//    def 'productType only accepts valid values'() {
+//        when:
+//        def ext = new BaseDistributionExtension(project, project.getObjects())
+//        ext.setProductType("foobar")
```

Comment:
probably safe to delete this test because nobody is actually expected to call `setProductType` right? We could possible even make it package private?